openpype/pipeline/load/utils.py

```diff
@@ -286,6 +288,54 @@ def get_representation_context(representation):
     return context
 
 
+def get_representation_files(context, filepath):
```

Comment:
I feel the logic here is still off. Can't we rely on the data in `representation["files"]` or something alike?

```python
def get_representation_files(representation, anatomy):
    
    context = representation["context"]
    files = []
    for file_data in representation["files"]:
        path = file_data["path"].format(root=anatomy.roots)
        files.append(path)
    return files
```

This would return all files **including** resources (that are not necessarily part of the sequence, e.g. for maya looks I believe this should also include the textures aisde of the maya ascii file). Filtering those might be done by matching the paths against `representation["data"]["template"]` to see if it's a match or not or reversing it and computing all files based on that:
```python
import copy
from openpype.lib import StringTemplate

def get_representation_frame_files(representation, anatomy, frames):
    
    template = representation["data"]["template"]
    template = StringTemplate(template)
    
    context = representation["context"]
    context = copy.deepcopy(context)
    context["root"] = anatomy.roots
    
    files = []
    for frame in frames:
        context["frame"] = frame
        path = template.format(**context)
        files.append(path)
        
    return files
```

Reply:
Not sure how well the last formatting example works with publishes that use `source` as template because it doesn't include `{frame}` I believe.

We might be able to capture those case to by doing:
```python
if "frame" not in context:
    raise ValueError("Can't resolve frame files for published contexts without frame data")

if "{frame" not in template:
    raise ValueError("Can't resolve frame files for publishes without {frame} key in template")
```

Reply:
For those edge cases we might be ok with falling back to a straight up parsing the files as a collection logic? Maybe even using `clique.assemble`?

Reply:
I do agree with the comments, but for sake of this PR please keep this function only inside resolve for now. This function definetelly should not be in public functions, at least not with this implementation.

Reply:
Thanks, good point - unless we can define an obvious 99% always correct function to retrieve this let's keep it Resolve lib only - so that we have some wiggle room to have it 'unstable' to some extent :P